pallets/pool-system/src/lib.rs

```diff
@@ -255,6 +259,51 @@ impl<CurrencyId, TrancheCurrency, EpochId, Balance, Rate, MetaSize, Weight, Tran
 		self.epoch.last_executed += One::one();
 		Ok(())
 	}
+
+	pub fn essence<T: Config>(&self) -> Result<PoolEssenceOf<T>, DispatchError>
```

Comment:
For simplicity do: 

`<T: Config<CurrencyId = CurrencyId, Balance = Balance, TrancheCurrency = TrancheCurrency, Rate = Rate>`